src/components/MerchantsPage/gam/CampaignProgressBar.tsx

```diff
@@ -0,0 +1,135 @@
+import * as React from 'react';
+import styled from 'styled-components';
+
+interface Props {
+  numContributions: number;
+  targetAmount: number;
+  progressBarColor: string;
+  lastContributionTime: Date;
+  endDate: Date;
+}
+
+const CampaignProgressBar = ({
+  numContributions,
+  targetAmount,
+  progressBarColor,
+  lastContributionTime,
+  endDate,
+}: Props) => {
+  const timeSinceLastContribution = Date.now() - lastContributionTime.getTime();
+  const minutesSinceLastConstribution = timeSinceLastContribution / (1000 * 60);
```

Comment:
Maybe better to have these as props on one object? Also there is a [`getMinutes()`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Date/getMinutes) function on date, not sure if that's what you need.

Would also avoid the typo here 😂 "con**st**ribution"

Reply:
Or actually just refactor this whole logic maybe to a separate function as it seems you're only using this to created a formatted string. See my other comment below.